corehq/apps/hqcase/case_helper.py

```diff
@@ -0,0 +1,188 @@
+from dataclasses import dataclass
+
+from corehq.apps.users.models import CouchUser
+
+from .api.updates import handle_case_update
+
+
+class CaseHelper:
+    """
+    CaseHelper aims to offer a simple interface for simple operations on
+    cases.
+
+    Initialize ``CaseHelper`` with an existing case ... ::
+
+        domain = 'hawaiian-mythology'
+        helper = CaseHelper(case=case, domain=domain)
+
+    ... or with a case ID::
+
+        helper = CaseHelper(case_id=case_id, domain=domain)
+
+    Or create a case using data in the format returned by the Case API
+    v0.6 ``serialize_case()`` function::
+
+        helper = CaseHelper(domain=domain)
+        helper.create_case({
+            'domain': 'hawaiian-mythology',
```

Comment:
Is this one necessary since `domain` was passed to the constructor?

Related, but not quite the same, would it raise an error if the domains were different?

Reply:
:thinking: Maybe I should drop that, to simplify the example.

It's not necessary. In fact, `_copy_case_data()` will drop it.